yarn-project/end-to-end/src/e2e_p2p_network.test.ts

```diff
@@ -70,6 +103,61 @@ describe('e2e_p2p_network', () => {
     await bootstrapNode.stop();
   });
 
+  it('should re-discover stored peers without bootstrap node', async () => {
+    const contexts: NodeContext[] = [];
+    const nodes: AztecNodeService[] = [];
+    for (let i = 0; i < NUM_NODES; i++) {
+      const node = await createNode(i + 1 + BOOT_NODE_UDP_PORT, bootstrapNodeEnr, i, `./data-${i}`);
+      nodes.push(node);
+    }
+    // wait a bit for peers to discover each other
+    await sleep(3000);
+
+    // stop bootstrap node
+    await bootstrapNode.stop();
+
+    // create new nodes from datadir
+    const newNodes: AztecNodeService[] = [];
+
+    // stop all nodes
+    for (let i = 0; i < NUM_NODES; i++) {
```

Comment:
Wouldn't it be better if we stop all nodes, and then restart them all?

Reply:
I felt this was a more "realistic" scenario of single nodes going down and restarted.
It can work with stopping all & restarting but seems to take quite a bit longer for them to re-discover, had to set the timeout to 30+ seconds